python/cudf/cudf/core/column/struct.py

```diff
@@ -65,6 +67,17 @@ def to_pandas(self, index: pd.Index = None, **kwargs) -> "pd.Series":
             pd_series.index = index
         return pd_series
 
+    @cached_property
+    def memory_usage(self):
+        n = 0
+        if self.nullable:
+            n += cudf._lib.null_mask.bitmask_allocation_size_bytes(self.size)
```

Comment:
~I think this is unsafe. `self.nullable` is not a once-only property so if we ask for this information _before_ a null mask has been set and subsequently modify the column to add a null, we won't notice and will under-report the subsequent memory usage.~

Edit: ah no, it's fine because `self.set_base_mask` eventually calls `del self.memory_usage`

Can we test this behaviour in an added test?

Reply:
Updated the existing tests to cover this 👍